components/isceobj/Alos2Proc/runIonSubband.py

```diff
@@ -296,30 +300,38 @@ def runIonSubband(self):
-                                      -4.0 * np.pi * secondaryTrack.frames[i].swaths[j].startingRange * (1.0/radarWavelength - 1.0/subbandRadarWavelength[k])
-                    if referenceTrack.frames[i].swaths[j-1].startingRange - secondaryTrack.frames[i].swaths[j-1].startingRange == \
-                       referenceTrack.frames[i].swaths[j].startingRange - secondaryTrack.frames[i].swaths[j].startingRange:
-                        #phaseDiff.append(phaseDiffSwath2 - phaseDiffSwath1)
-                        #if reference and secondary versions are all before or after version 2.025 (starting range error < 0.5 m), 
-                        #it should be OK to do the above.
-                        #see results in neom where it meets the above requirement, but there is still phase diff
-                        #to be less risky, we do not input values here
-                        phaseDiff.append(None)
-                    else:
-                        phaseDiff.append(None)
+                # #compute phase needed to be compensated using startingRange
+                # if j >= 1:
+                #     #phaseDiffSwath1 = -4.0 * np.pi * (referenceTrack.frames[i].swaths[j-1].startingRange - secondaryTrack.frames[i].swaths[j-1].startingRange)/subbandRadarWavelength[k]
+                #     #phaseDiffSwath2 = -4.0 * np.pi * (referenceTrack.frames[i].swaths[j].startingRange - secondaryTrack.frames[i].swaths[j].startingRange)/subbandRadarWavelength[k]
+                #     phaseDiffSwath1 = +4.0 * np.pi * referenceTrack.frames[i].swaths[j-1].startingRange * (1.0/radarWavelength - 1.0/subbandRadarWavelength[k]) \
+                #                       -4.0 * np.pi * secondaryTrack.frames[i].swaths[j-1].startingRange * (1.0/radarWavelength - 1.0/subbandRadarWavelength[k])
+                #     phaseDiffSwath2 = +4.0 * np.pi * referenceTrack.frames[i].swaths[j].startingRange * (1.0/radarWavelength - 1.0/subbandRadarWavelength[k]) \
+                #                       -4.0 * np.pi * secondaryTrack.frames[i].swaths[j].startingRange * (1.0/radarWavelength - 1.0/subbandRadarWavelength[k])
+                #     if referenceTrack.frames[i].swaths[j-1].startingRange - secondaryTrack.frames[i].swaths[j-1].startingRange == \
+                #        referenceTrack.frames[i].swaths[j].startingRange - secondaryTrack.frames[i].swaths[j].startingRange:
+                #         #phaseDiff.append(phaseDiffSwath2 - phaseDiffSwath1)
+                #         #if reference and secondary versions are all before or after version 2.025 (starting range error < 0.5 m), 
+                #         #it should be OK to do the above.
+                #         #see results in neom where it meets the above requirement, but there is still phase diff
+                #         #to be less risky, we do not input values here
+                #         phaseDiff.append(None)
+                #     else:
+                #         phaseDiff.append(None)
```

Comment:
Commented code rots easily, so you can go ahead and remove it. It will be retained in the git history if we ever need it again.

Reply:
Understand, but they are important codes for testing issues relating to ALOS-2. ALOS-2 has a lot of issues. I would just leave them here for now. I may still need them occasionally. I can delete them when I think they are no longer needed.

Reply:
In that case, how about putting them in an `if False:` block? That way we can at least be sure they are parsed correctly by python.

Reply:
updated.